### PR TITLE
[DQL3-4] Currency email notification

### PR DESCRIPTION
**Pull request by tino097**
Add js script to handle `next_update_due` and `nature_of_change` fields
Update templates with required changes for showing metadata for authorized users

### CODE REVIEW

**Comment by duttonw** [comment]
some tests are failing on last build, would suggest page failed to load successfully.

```
Failing scenarios:
  test/features/google_analytics.feature:8  When viewing the HTML source code of a dataset, the organisation meta data is visible
  test/features/google_analytics.feature:13  When viewing the HTML source code of a resource, the organisation meta data is visible

4 features passed, 1 failed, 0 skipped
14 scenarios passed, 2 failed, 0 skipped
54 steps passed, 2 failed, 1 skipped, 0 undefined
Took 1m3.573s
```

File: ckanext/data_qld_theme/fanstatic/next-update-due.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+jQuery(document).ready(function () {
+
+    UPDATE_FREQUENCY_DAYS = JSON.parse(jQuery('input[name="update_frequency_days"]').val())
```
Comment (MarkCalvert) [inline ckanext/data_qld_theme/fanstatic/next-update-due.js:3]: Should we set a default value of a empty dict just incase there is no value in the input field etc?

File: ckanext/data_qld_theme/fanstatic/resource_file_upload.js [hunk 2]
```diff
@@ -1,26 +1,42 @@
 jQuery(document).ready(function () {
```
Comment (MarkCalvert) [inline ckanext/data_qld_theme/fanstatic/resource_file_upload.js:1]: Add some general comments to help explain the business logic and reasons
Remove extra blank lines etc
Run lint and format the code

File: ckanext/data_qld_theme/fanstatic/resource_file_upload.js [hunk 3]
```diff
@@ -1,26 +1,42 @@
...
   else if (jQuery('.image-upload[data-module="image-upload"]').data('module-is_url') == true) {
     jQuery('#field-size').parent().parent().show();
+    
   }
 
   // Hide the resource element if a file was selected to upload
   jQuery('#field-image-upload').change(function () {
     jQuery('#field-size').parent().parent().fadeOut();
+
+    if (jQuery('input[name="id"]').val().length > 0) {
```
Comment (MarkCalvert) [inline ckanext/data_qld_theme/fanstatic/resource_file_upload.js:19]: Add comment explaining why you are checking for this element and the logic

File: ckanext/data_qld_theme/fanstatic/next-update-due.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+jQuery(document).ready(function () {
```
Comment (MarkCalvert) [inline ckanext/data_qld_theme/fanstatic/next-update-due.js:1]: Add some general comments to help explain the business logic and reasons
Remove extra blank lines etc
Run lint and format the code

File: ckanext/data_qld_theme/templates/package/snippets/additional_info.html [hunk 5]
```diff
@@ -12,6 +12,7 @@
 
 {%- if h.data_qld_user_has_admin_access(True) == False -%}
   {%- do exclude_fields.append('author_email') -%}
+  {%- do exclude_fields.append('next_update_due') -%}
```
Comment (MarkCalvert) [inline ckanext/data_qld_theme/templates/package/snippets/additional_info.html:15]: Can any user with editor or admin access to any organisation view this metadata field or is it only users with editor/admin access to this datasets organisation?
If it is the later do a check like `data_qld_show_de_identified_data`

File: ckanext/data_qld_theme/templates/scheming/form_snippets/update_due.html [hunk 6]
```diff
@@ -0,0 +1,21 @@
+
```
Comment (MarkCalvert) [inline ckanext/data_qld_theme/templates/scheming/form_snippets/update_due.html:1]: If this is an exact copy of the scheming date form snippet just call the snippet directly.
eg. {%- snippet 'scheming/form_snippets/date.html', field=field, data=data, errors=errors -%}

**Review by MarkCalvert — CHANGES_REQUESTED** [review]

**Review by MarkCalvert — APPROVED** [review]

**Pull request merged**